rest-service/manager_rest/storage/storage_manager.py

```diff
@@ -475,18 +475,16 @@ def _validate_pagination(pagination_size):
             )
 
     @no_autoflush
-    def _validate_unique_resource_id_per_tenant(self, instance):
+    def _validate_unique_resource_per_tenant(self, instance):
         """Assert that only a single resource exists with a given id in a
         given tenant
         """
         # Only relevant for resources that have unique IDs and are connected
         # to a tenant
-        if not instance.is_resource or not instance.is_id_unique:
+        if not instance.is_resource or not instance.check_unique_query():
             return
 
-        query = self._get_unique_resource_id_query(instance.__class__,
-                                                   instance.id,
-                                                   instance.tenant)
+        query = self._get_unique_resource_query(instance)
```

Comment:
it'd be good to only get the query once (ie. only call `check_unique_query` once), instead of here, and in `_get_unique_resource_query` as well

I'm thinking just something like `if not is_resource: return;  query = instance.check_query();  if not query: return`

Then, `_get_unique_resource_query` can just be inlined here, or even into the base class method